core/src/main/scala/org/bitcoins/core/wallet/fee/FeeUnit.scala

```diff
@@ -1,47 +1,43 @@
 package org.bitcoins.core.wallet.fee
 
-import org.bitcoins.core.currency.{CurrencyUnit, CurrencyUnits, Satoshis}
+import org.bitcoins.core.currency.{CurrencyUnit, Satoshis}
 import org.bitcoins.core.protocol.transaction.Transaction
 
 /**
   * This is meant to be an abstract type that represents different fee unit measurements for
   * blockchains
   */
 sealed abstract class FeeUnit {
-  def currencyUnit: CurrencyUnit
   def *(tx: Transaction): CurrencyUnit = calc(tx)
-  def calc(tx: Transaction): CurrencyUnit = Satoshis(tx.vsize * toLong)
-  def toLong: Long = currencyUnit.satoshis.toLong
+  def calc(tx: Transaction): CurrencyUnit
+  def baseAmount: Double
 }
 
 /**
   * Meant to represent the different fee unit types for the bitcoin protocol
   * @see [[https://en.bitcoin.it/wiki/Weight_units]]
   */
-sealed abstract class BitcoinFeeUnit extends FeeUnit
+sealed abstract class BitcoinFeeUnit extends FeeUnit {
+  def sats: Double
```

Comment:
So I don't think it makes sense to have `sats: Double` as you can't have sats be a `Double` in the bitcoin protocol. 

If you go to other projects, and they allow you to select a fee, do they let you select `1.5` satoshis/vbyte as a option? 

There seems to be a `desiredFee` and `actualFee` or something like that.

Reply:
Thinking about this more, it seems what we don't represent well is a fee-rate, rather than an absolute fee. An absolute fee cannot be a double (at least in the sats unit). However a fee rate most definitely can.

It seems to me we are using the `FeeUnit` type for both at the moment, when it reality we maybe should have a `FeeUnit` type and a `FeeRate` type? Thoughts?

Reply:
> If you go to other projects, and they allow you to select a fee, do they let you select 1.5 satoshis/vbyte as a option?

I know electrum and wasabi allow you to, I haven't used many other wallets.